metsim/methods/mtclim.py

```diff
@@ -19,197 +19,80 @@
-        Elevation in meters
-    params:
-        A dictionary of parameters from the
-        MetSim object
-    """
-    def _calc_tfmax(prec, dtr, sm_dtr):
-        """Estimate cloudy day transmittance"""
-        b = cnst.B0 + cnst.B1 * np.exp(-cnst.B2 * sm_dtr)
-        t_fmax = 1.0 - 0.9 * np.exp(-b * np.power(dtr, cnst.C))
-        inds = np.array(prec > params['sw_prec_thresh'])
-        t_fmax[inds] *= params['rain_scalar']
-        return t_fmax
-
-    # Calculate the diurnal temperature range
-    df['t_max'] = np.maximum(df['t_max'], df['t_min'])
-    dtr = df['t_max'] - df['t_min']
-    df['dtr'] = dtr
-    sm_dtr = df['smoothed_dtr']
-    df['tfmax'] = _calc_tfmax(df['prec'], dtr, sm_dtr)
-    tdew = df.get('tdew', df['t_min'])
-    pva = df.get('hum', svp(tdew.values))
-    pa = atm_pres(elev, params['lapse_rate'])
-    yday = df.index.dayofyear - 1
-    df['dayl'] = sg['daylength'][yday]
-
-    # Calculation of tdew and shortwave. tdew is iterated on until
-    # it converges sufficiently
-    tdew_old = tdew
-    tdew, pva = sw_hum_iter(df, sg, pa, pva, dtr, params)
-    while(np.sqrt(np.mean((tdew-tdew_old)**2)) > params['tdew_tol']):
-        tdew_old = np.copy(tdew)
-        tdew, pva = sw_hum_iter(df, sg, pa, pva, dtr, params)
-    df['vapor_pressure'] = pva
-
-
-def sw_hum_iter(df: pd.DataFrame, sg: dict, pa: float, pva: pd.Series,
-                dtr: pd.Series, params: dict):
-    """
-    Calculated updated values for dewpoint temperature
-    and saturation vapor pressure.
-
-    Parameters
-    ----------
-    df:
-        Dataframe containing daily timeseries of
-        cloud cover fraction, tfmax, swe, and
-        shortwave radiation
-    sg:
-        Solar geometry dictionary, calculated with
-        `metsim.physics.solar_geom`.
-    pa:
-        Air pressure in Pascals
-    pva:
-        Vapor presure in Pascals
-    dtr:
-        Daily temperature range
-    params:
-        A dictionary of parameters from a MetSim object
-
-    Returns
-    -------
-    (tdew, svp):
-        A tuple of dewpoint temperature and saturation
-        vapor pressure
-    """
-    tt_max0 = sg['tt_max0']
-    potrad = sg['potrad']
-    daylength = sg['daylength']
-    yday = df.index.dayofyear - 1
-
-    t_tmax = np.maximum(tt_max0[yday] + (cnst.ABASE * pva), 0.0001)
-    t_final = t_tmax * df['tfmax']
-    df['pva'] = pva
-    df['tfinal'] = t_final
-
-    # Snowpack contribution
-    sc = np.zeros_like(df['tfmax'])
-    if params['mtclim_swe_corr']:
-        inds = np.logical_and(df['swe'] > 0.,  daylength[yday] > 0.)
-        sc[inds] = ((1.32 + 0.096 * df['swe'][inds]) *
-                    1.0e6 / daylength[yday][inds])
-        sc = np.maximum(sc, 100.)
-
-    # Calculation of shortwave is split into 2 components:
-    # 1. Radiation from incident light
-    # 2. Influence of snowpack - optionally set by MTCLIM_SWE_CORR
-    df['shortwave'] = potrad[yday] * t_final + sc
-
-    # Calculate cloud effect
+def run(df, params):
```

Comment:
@arbennett - do you know why the mtclim changes are here in this PR?

Reply:
Actually, it looks like there are some conflicts to resolve. Let me know once those are fixed.